c_binding/lsm_datatypes.cpp

```diff
@@ -874,6 +876,28 @@ int lsm_system_mode_get(lsm_system *s, lsm_system_mode_type *mode)
         return LSM_ERR_NO_SUPPORT;
 }
 
+int lsm_system_read_cache_pct_set(lsm_system *sys, int read_pct)
+{
+    if ((sys == NULL) || (! LSM_IS_SYSTEM(sys)) ||
+        (read_pct == LSM_SYSTEM_CACHE_PCT_NO_SUPPORT))
+        return LSM_ERR_INVALID_ARGUMENT;
+
+    sys->read_cache_pct = read_pct;
+
+    return LSM_ERR_OK;
+}
+
+int lsm_system_read_cache_pct_get(lsm_system *s, int *read_pct)
+{
+    if ((s == NULL) || (read_pct == NULL) || (! LSM_IS_SYSTEM(s)))
+        return LSM_ERR_INVALID_ARGUMENT;
+
+    if (s->read_cache_pct != LSM_SYSTEM_CACHE_PCT_NO_SUPPORT)
+        *read_pct = s->read_cache_pct;
+
```

Comment:
Ah, I misunderstood which read_pct check you were talking about (both showed up in the code snapshot I saw in email, and I chose the first check rather than the second).

Will fix as you suggested for the gets.

Joe

On Apr 6, 2016, at 9:35 AM, Tony Asleson <notifications@github.com<mailto:notifications@github.com>> wrote:


In c_binding/lsm_datatypes.cpp<https://github.com/libstorage/libstoragemgmt/pull/102#discussion_r58715718>:

> +        (read_pct == LSM_SYSTEM_CACHE_PCT_NO_SUPPORT))
> +        return LSM_ERR_INVALID_ARGUMENT;
> +
> +    sys->read_cache_pct = read_pct;
> +
> +    return LSM_ERR_OK;
> +}
> +
> +int lsm_system_read_cache_pct_get(lsm_system *s, int *read_pct)
> +{
> +    if ((s == NULL) || (read_pct == NULL) || (! LSM_IS_SYSTEM(s)))
> +        return LSM_ERR_INVALID_ARGUMENT;
> +
> +    if (s->read_cache_pct != LSM_SYSTEM_CACHE_PCT_NO_SUPPORT)
> +        *read_pct = s->read_cache_pct;
> +


On 04/06/2016 09:16 AM, Joe Handzik wrote: We could do that. We'll need to change the behavior for fw_version and mode set too, that's what I based my implementation on.
I'm confused by this statement as I see this for the implementation of the 'get' operations for fw version and mode, both of which will return LSM_ERR_NO_SUPPORT? int lsm_system_fw_version_get(lsm_system *s, const char **fw_ver) { if ((s == NULL) || (fw_ver == NULL) || (! LSM_IS_SYSTEM(s))) return LSM_ERR_INVALID_ARGUMENT; *fw_ver = s->fw_version; if (s->fw_version[0] != '\0' ) return LSM_ERR_OK; else return LSM_ERR_NO_SUPPORT; } int lsm_system_mode_get(lsm_system *s, lsm_system_mode_type *mode) { if ((s == NULL) || (mode == NULL) || (! LSM_IS_SYSTEM(s))) return LSM_ERR_INVALID_ARGUMENT; *mode = s->mode; if (s->mode != LSM_SYSTEM_MODE_NO_SUPPORT) return LSM_ERR_OK; else return LSM_ERR_NO_SUPPORT; } Regards, Tony

-
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/libstorage/libstoragemgmt/pull/102/files/d382741c3ecc2d9dcc6442563e3af2dc427181f6#r58715718>